ws-nextjs-app/pages/[service]/new_live/[id]/[[...variant]].page.tsx

```diff
@@ -28,29 +31,66 @@ interface PageDataParams extends ParsedUrlQuery {
   renderer_env?: string;
 }
 
+const logger = nodeLogger(__filename);
+
 const getPageData = async ({
   id,
   page,
   service,
   variant,
   rendererEnv,
 }: PageDataParams) => {
-  const data = await bffFetch({
-    getAgent,
+  const pathname = `${id}${rendererEnv ? `?renderer_env=${rendererEnv}` : ''}`;
+  const livePageUrl = constructPageFetchUrl({
     page,
-    path: `${id}${rendererEnv ? `?renderer_env=${rendererEnv}` : ''}`,
+    pageType: 'live',
+    pathname,
     service,
     variant,
   });
 
+  const env = getEnvironment(pathname);
+  const optHeaders = { 'ctx-service-env': env };
+  const isLocal = !env || env === 'local';
+
+  const agent = !isLocal ? await getAgent() : null;
+
+  let pageStatus;
+  let pageJson;
+  let errorMessage;
+
+  const path = livePageUrl.toString();
+
+  try {
+    // @ts-expect-error Due to jsdoc inference, and no TS within fetchPageData
+    const { status, json } = await fetchPageData({
+      path,
+      agent,
+      optHeaders,
+    });
+    pageStatus = status;
+    pageJson = json;
+  } catch ({ message, status }) {
```

Comment:
When I check this out locally, I get an errors `Property 'message' does not exist on type 'unknown'` and `'Property 'status' does not exist on type 'unknown'.` I'm assuming this is just Typescript flagging that types have not been defined and the code still works. However is there a fix for this we should use going forward?

Reply:
Yeah I saw that, those errors also exist [here](https://github.com/bbc/simorgh/blob/db731779c8b3f9a7a9ff759ea7eb9ee867262fc2/src/app/routes/article/getInitialData/index.ts#L144) and [here](https://github.com/bbc/simorgh/blob/db731779c8b3f9a7a9ff759ea7eb9ee867262fc2/src/app/routes/homePage/getInitialData/index.ts#L65), which is where the logic has been reused. Not sure how to fix this in TS